### PR TITLE
val metrics storing,

### PR DESCRIPTION
**Pull request by sicotfre**
ChangeLog:
- validation metrics are computed as soon as a validation set is provided (regardless of the `early_stopping` argument)
- store validation metrics during epochs iterations in `SVD.metrics_` (I love to monitor algo convergence :). Retrieve a Pandas DataFrame with `SVD.get_val_metrics()` 
- add `min_delta` argument in `fit()`
- some picky syntax cleaning

### CODE REVIEW

**Comment by sicotfre** [comment]
Travis fails but it is due to `urllib` in the `dataset` module which I left unchanged.

I believe the proper way to import it is:
```
import urllib.request
with urllib.request.urlopen('http://python.org/') as response:
   html = response.read()
```

https://docs.python.org/3.7/howto/urllib2.html

*edit:* I made the change just to check myself

File: funk_svd/__init__.py [hunk 1]
```diff
@@ -2,4 +2,4 @@
 from .svd import SVD
 from . import utils
 
-__version__ = '0.0.1.dev1'
+__version__ = '0.0.1.dev2'
```
Comment (gbolmier) [inline funk_svd/__init__.py]: I just moved the lib version to a new file `funk_svd/__version__.py`, could you can move the change there?

**Comment by gbolmier** [comment]
Thanks for the PR @sicotfre!

Monitoring is a very good idea! I reviewed the changes, it looks good to me. Let me know when you consider the PR ready to be merged :)

File: funk_svd/__init__.py (hunk 1 above)
Comment (sicotfre) [reply funk_svd/__init__.py]: actually I'll just let you manage the version the way you want.

**Comment by sicotfre** [comment]
ok to merge!

**Pull request merged**